apps/homescreen/js/pages.js

```diff
@@ -219,9 +219,10 @@
       // Remove child immediately, datastore operations can be quite slow
       this.removeCard(this.selectedCard);
 
+      var revId = this.pagesStore.datastore.revisionId;
       this.pagesStore.get(id).then(entry => {
         entry.data.pinned = false;
-        this.pagesStore.datastore.put(entry.data, id).then(() => {},
+        this.pagesStore.datastore.put(entry.data, id, revId).then(() => {},
```

Comment:
Sorry, I said that wrong, it doesn't have to be the same URL - any simultaneous DataStore update will cause a versioning conflict. I'll see if I can improve my patch so that it shows a warning to the user ('Unpinning failed, please try again'), and also does not remove the pinned page from the container if it was not really removed.